frontend/packages/knative-plugin/src/components/traffic-splitting/__tests__/TrafficSplitting.spec.tsx

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import * as Renderer from 'react-test-renderer';
+import TrafficSplitting from '../TrafficSplitting';
+import {
+  mockRevisions,
+  mockServiceData,
+} from '../../../utils/__mocks__/traffic-splitting-utils-mock';
+
+describe('TrafficSplitting', () => {
+  it('should match the previous traffic splitting snapshot', () => {
+    const tree = Renderer.create(
+      <TrafficSplitting service={mockServiceData} revisions={mockRevisions} />,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
```

Comment:
Why are you using snapshot testing here? I don't think snapshot testing is the right way here because If there are some changes made in the `TrafficSplittingModal` component this test will fail even though there are no changes made to the `TrafficSplitting` component.

Reply:
Yeah, I agree with Sahil. Snapshot testing isn't very useful in this case. I usually don't prefer snapshot testing. You should be testing some rendering logic of the component or even the validation logic.

Reply:
Snapshot testing this is not useful at all.
We should be using `shallow` to mount and then assert specific changes in in the rendered output.

Reply:
removed snapshot testing and updated code with shallow testing